dkeyserver/Enclave/enclave.cpp

```diff
@@ -387,7 +335,333 @@ int handle_communication_until_done(
     return ret;
 }
 
-int sgx_set_up_tls_server(char *server_port)
+static unsigned long inet_addr(const char *str)
+{
+    unsigned long lHost = 0;
+    char *pLong = (char *)&lHost;
+    char *p = (char *)str;
+    while (p)
+    {
+        *pLong++ = atoi(p);
+        p = strchr(p, '.');
+        if (p)
+            ++p;
+    }
+    return lHost;
+}
+
+int connect(int sockfd, const struct sockaddr *servaddr, socklen_t addrlen)
+{
+    int ret = -1;
+
+    if (ocall_connect(&ret, sockfd, servaddr, addrlen) == SGX_SUCCESS)
+        return ret;
+
+    return -1;
+}
+
+int create_socket(const char *server_name, uint16_t server_port)
+{
+    int sockfd = -1;
+    struct sockaddr_in dest_sock;
+    int ret = -1;
+
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    if (sockfd == -1)
+    {
+        log_d(TLS_SERVER "Error: Cannot create socket %d.\n", errno);
+        goto out;
+    }
+
+    dest_sock.sin_family = AF_INET;
+    dest_sock.sin_port = htons(server_port);
+    dest_sock.sin_addr.s_addr = inet_addr(server_name);
+    bzero(&(dest_sock.sin_zero), sizeof(dest_sock.sin_zero));
+
+    if (connect(
+            sockfd, (sockaddr *)&dest_sock,
+            sizeof(struct sockaddr)) == -1)
+    {
+        log_d(
+            TLS_SERVER "failed to connect to target server %d:%d (errno=%d)\n",
+            server_name,
+            server_port,
+            errno);
+        ocall_close(&ret, sockfd);
+        if (ret != 0)
+            log_d(TLS_SERVER "OCALL: error closing socket\n");
+        sockfd = -1;
+        goto out;
+    }
+    log_d(TLS_SERVER "connected to target server %s:%d\n", server_name, server_port);
+
+out:
+    return sockfd;
+}
+
+sgx_status_t get_domainkey_from_target(uint8_t *domain_key, const char *target_server_name, uint16_t target_server_port)
+{
+    log_i(TLS_SERVER "Connecting target server...\n");
+    sgx_status_t ret = SGX_ERROR_UNEXPECTED;
+
+    if (domain_key != NULL){
+        log_i("domain key already exist. \n");
+        return ret;
+    }
+    if (target_server_name[0] == '\0' && target_server_port == 0){
+        log_i("no target server. \n");
+        return ret;
+    }
+
+    SSL_CTX *ssl_client_ctx = nullptr;
+    SSL *ssl_session = nullptr;
+
+    X509 *cert = nullptr;
+    EVP_PKEY *pkey = nullptr;
+    SSL_CONF_CTX *ssl_confctx = SSL_CONF_CTX_new();
+
+    int client_socket = -1;
+    int error = 0;
+    unsigned char buf[200];
+    int len = 0;
+    int bytes_written = 0;
+    int bytes_read = 0;
+
+    if (target_server_name == NULL)
```

Comment:
move to line 411:
if (target_server_name == NULL || target_server_name[0] == '\0')
{
log;
return;
}

Reply:
done.